lib/matplotlib/contour.py

```diff
@@ -354,7 +355,10 @@ def locate_label(self, linecontour, labelwidth):
         xlast = XX[:, -1].reshape(xsize, 1)
         s = (yfirst - YY) * (xlast - xfirst) - (xfirst - XX) * (ylast - yfirst)
         L = np.sqrt((xlast - xfirst) ** 2 + (ylast - yfirst) ** 2).ravel()
-        dist = np.add.reduce(([(abs(s)[i] / L[i]) for i in range(xsize)]), -1)
+        # Ignore warning that divide by zero throws, as this is a valid option
+        with np.errstate(divide='ignore', invalid='ignore'):
+            dist = np.add.reduce(([(abs(s)[i] / L[i]) for i in range(xsize)]),
```

Comment:
I have no idea - I tried to understand what was going on here for a while but gave up...